docs (domain/OIDC): add domain docs for OIDC auth method attributes (#3490)
hashicorp/boundary: website/content/docs/concepts/domain-model/auth-methods.mdx

- `min_password_length` - (required) The default is 8.

### OIDC auth method attributes

The OIDC auth method has the following additional attributes:

> - `account_claim_maps` (optional list) These are a map from custom
>   claims to the standard claims of sub, name and email. These maps are
>   represented as key=value where the key equals the Provider from-claim and the
>   value equals the Boundary to-claim. For example "oid=sub". May be specified
>   multiple times for different to-claims.

Reviewer: ```suggestion
- `account_claim_maps` (optional list) These are a map from custom
  claims to the standard claims of sub, name, and email. These maps are
  represented as key=value where the key equals the provider from-claim and the
  value equals the Boundary to-claim. For example "oid=sub". You can specify this attribute
  multiple times for different to-claims.
```

> - `allowed_audiences` (optional list) Audiences for which the provider
>   responses will be allowed

Reviewer: ```suggestion
- `allowed_audiences` (optional list) Audiences for which provider
  responses are allowed.
```

> - `api_url_prefix` (required) The API prefix to use when generating callback URLs
>   for the provider. Should be set to an address at which the provider can reach
>   back to the controller.

Reviewer: ```suggestion
- `api_url_prefix` (required) The API prefix to use when generating callback URLs
  for the provider. You should set the value to an address that allows the provider to reach 
  the controller.
```

- `callback_url` (output read-only) The URL that should be provided to the IdP
  for callbacks.

> - `claims_scopes` (optional list) The claims scope requested. May be specified
>   multiple times.

Reviewer: ```suggestion
- `claims_scopes` (optional list) The claims scope requested. You can specify this attribute
  multiple times.
```

> - `client_id` (required) The OAuth 2.0 Client Identifier this auth method should
>   use with the provider.

Reviewer: ```suggestion
- `client_id` (required) The OAuth 2.0 client identifier this auth method should
  use with the provider.
```

- `client-secret` (required) The corresponding client secret.
  

> - `client_secret_hmac` (output read-only) The HMAC of the client secret returned
>   by the Boundary controller, which is used for comparison after initial setting
>   of the value.

Reviewer: ```suggestion
- `client_secret_hmac` (output read-only) The HMAC of the client secret that the Boundary controller
  returns. It is used for comparison to the value's initial setting.
```

> - `description` - (optional)

Reviewer: I think we should delete `description` since it is include in the first section as one of the attributes that is common to all auth methods.

> - `disable_discovered_config_validation` (optional) Disables validation logic
>   ensuring that the OIDC provider's information from its discovery endpoint
>   matches the information here. The validation is only performed at create or
>   update time.

Reviewer: ```suggestion
- `disable_discovered_config_validation` (optional) Disables validation logic
  to ensure that the OIDC provider's information from its discovery endpoint
  matches the information here. The validation is only performed at create or
  update time.
```

> - `idp_ca_certs` - (optional) PEM-encoded X.509 CA certificate that can be used
>   as trust anchors when connecting to an OIDC provider. May be specified
>   multiple times.

Reviewer: ```suggestion
- `idp_ca_certs` - (optional) PEM-encoded X.509 CA certificate that can be used
  as trust anchors when you connect to an OIDC provider. You can specify this attribute
  multiple times.
```

> - `issuer` - (required) The provider's Issuer URL. which must match the issuer
>   field in generated tokens.

Reviewer: ```suggestion
- `issuer` - (required) The provider's issuer URL. This value must match the issuer
  field in generated tokens.
```

> - `max_age` (optional) The max age to provide to the provider, indicating how
>   much time is allowed to have passed since the last authentication before the
>   user is challenged again. A value of 0 sets an immediate requirement for all
>   users to reauthenticate, and an unset maxAge results in a Terraform value of
>   -1 and the default TTL of the chosen OIDC will be used.

Reviewer: ```suggestion
- `max_age` (optional) The max age to send to the provider. This value indicates how
  much time is allowed to have passed since the last authentication before the
  user is challenged again. A value of `0` sets an immediate requirement for all
  users to reauthenticate, and an unset `maxAge` results in a Terraform value of
  -1 and the default TTL of the chosen OIDC is used.
```

> - `name` (optional) The auth-method name.

Reviewer: Same as `description`, I think we should delete the entry for `name` here, since it's listed at the top of this page as an attribute that all auth methods have in common.
